lib/tlscontext.c

```diff
@@ -334,7 +335,25 @@ tls_context_setup_session(TLSContext *self)
         }
 
       SSL_CTX_set_verify(self->ssl_ctx, verify_mode, tls_session_verify_callback);
-      SSL_CTX_set_options(self->ssl_ctx, SSL_OP_NO_SSLv2);
+      if (self->ssl_options != TSO_NONE)
+        {
+          ssl_options=0;
+          if(self->ssl_options & TSO_NOSSLv2)
+            ssl_options |= SSL_OP_NO_SSLv2;
+          if(self->ssl_options & TSO_NOSSLv3)
+            ssl_options |= SSL_OP_NO_SSLv3;
+          if(self->ssl_options & TSO_NOTLSv1)
+            ssl_options |= SSL_OP_NO_TLSv1;
+#ifdef SSL_OP_NO_TLSv1_2
+          if(self->ssl_options & TSO_NOTLSv11)
+            ssl_options |= SSL_OP_NO_TLSv1_1;
+          if(self->ssl_options & TSO_NOTLSv12)
+            ssl_options |= SSL_OP_NO_TLSv1_2;
+#endif
+          SSL_CTX_set_options(self->ssl_ctx, ssl_options);
+        }
+      else
+	msg_debug("empty ssl options",NULL);
```

Comment:
On the long term, it may reduce the size of code, and would gave us the possibility to simply add new options.
Maybe. Maybe we want to gave the chance through our implementation, to not set, but instead SSL_CTX_clear_options regarding the default set.
Then it may almost double the size of your version of code if it's flexible enough at all.
In that case also harden our work, if you want to make two of syslog-ng's ssl-options option to be mutualy exclusive, etc.
So at this point I would not bother this type of solution, since at this point, with this feature set, this is the most compact version. In the future, if there is real need for other options, the it may have a worth to think about that, but I would say, not now, not with this simple feature set.